frontend/pages/auth/recovery.tsx

```
@@ -0,0 +1,83 @@
+import React from "react";
+import { GetServerSideProps } from "next";
+import { getRecoveryFields } from "../../lib/auth";
+import { sendEvent } from "../../lib/events";
+import { RequestMethodConfig } from "@oryd/kratos-client";
+
+type RecoveryProps = {
+  request: string;
+  messages: string[] | null;
+  formConfig: RequestMethodConfig;
+};
+
+export const getServerSideProps: GetServerSideProps = async (
+  context
+): Promise<{ props: RecoveryProps }> => {
+  const request = context.query.request;
+
+  if (!request && context.res) {
+    context.res.writeHead(302, {
+      Location: "/.ory/kratos/public/self-service/browser/flows/recovery",
+    });
+    context.res.end();
+    return { props: {} as any };
+  }
+  if (!request || Array.isArray(request)) {
+    return { props: {} as any };
+  }
```

I wonder what we want to do in this case. Can this actually happen? Should we throw an error?